### ISSUE TITLE
Unable to connect kuma dataplane with a control plane running on another machine

### ISSUE THREAD

**Issue by hakeemsyd** [kumahq/kuma#1193]
### Summary
I am starting a kuma controlplane on an ec2 instance on aws. The instance has a public ip address and i verified that controlplane has opened port 5678 (by telnet).
I am unable to start a dataplane that can connect to controlplane running on ec2 instance on aws.

### Steps To Reproduce

1. Start up a data plance on aws ec2 instance. 
2. Enable inbound connections to port 5678 on ec2 instance
3. Generate token on ec2 by running `kumactl generate dataplane-token  --mesh=default`. 
4. Copy the token to your computer and use it as inpu tto -dataplane-token-file on next step.
5. Start a dataplane on your laptop with `--cp-address=<public-address-of-ec2:5678>`
e.g. 
```
kuma-dp run \
--cp-address=https://ec2-18-144-4-220.us-west-1.compute.amazonaws.com:5678 \
--dataplane-token-file=/tmp/dp-token  \
--dataplane-file=universal/dp.yaml \
```

Dataplane prints error on terminal.
`[2020-11-22 17:53:40.178][2800877][warning][config] [bazel-out/darwin-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:93] StreamAggregatedResources gRPC config stream closed: 13,`

Error logs on controlplane
```
ERROR	kds-remote.mux-client	component terminated with an error	{"generationID": 1417, "error": "rpc error: code = Unavailable desc = transport is closing"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/kumahq/kuma/pkg/core/runtime/component.(*resilientComponent).Start
	/go/src/github.com/kumahq/kuma/pkg/core/runtime/component/resilient.go:51
github.com/kumahq/kuma/pkg/core/runtime/component.(*manager).startLeaderComponents.func2.1
	/go/src/github.com/kumahq/kuma/pkg/core/runtime/component/component.go:113
```
### Additional Details & Logs

- Version: 1.0.0
- Error logs: 
Contorlplane error logs:
```

ERROR	kds-remote.mux-client	component terminated with an error	{"generationID": 1417, "error": "rpc error: code = Unavailable desc = transport is closing"}
github.com/go-logr/zapr.(*zapLogger).Error
	/go/pkg/mod/github.com/go-logr/zapr@v0.1.0/zapr.go:128
github.com/kumahq/kuma/pkg/core/runtime/component.(*resilientComponent).Start
	/go/src/github.com/kumahq/kuma/pkg/core/runtime/component/resilient.go:51
github.com/kumahq/kuma/pkg/core/runtime/component.(*manager).startLeaderComponents.func2.1
	/go/src/github.com/kumahq/kuma/pkg/core/runtime/component/component.go:113
```

Dataplane error log:
`[2020-11-22 17:53:40.178][2800877][warning][config] [bazel-out/darwin-opt/bin/external/envoy/source/common/config/_virtual_includes/grpc_stream_lib/common/config/grpc_stream.h:93] StreamAggregatedResources gRPC config stream closed: 13,`

- Configuration
- Platform and Operating System: Mac OS 10.15.4, AWS EC2

**Comment by nickolaev** [comment]
This should be addressed already in the latest `1.0.3` version. I think it was related to the format of the (ever changing its IP) DNS name -> `https://ec2-18-144-4-220.us-west-1.compute.amazonaws.com:5678`

**Comment by nickolaev** [comment]
Closing, please re-open if it is still relevant.
